fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/shears/TripwireBlockMixin.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.item.shears;
+
+import com.llamalad7.mixinextras.injector.ModifyExpressionValue;
+import com.llamalad7.mixinextras.sugar.Local;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+
+import net.minecraft.block.TripwireBlock;
+import net.minecraft.entity.player.PlayerEntity;
+
+import net.fabricmc.fabric.api.tag.convention.v1.ConventionalItemTags;
+
+@Mixin(TripwireBlock.class)
+public abstract class TripwireBlockMixin {
+	@ModifyExpressionValue(at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isOf(Lnet/minecraft/item/Item;)Z"), method = "onBreak")
+	private boolean isShears(boolean original, @Local PlayerEntity player) {
+		// allows anything in fabric:shears to silently break tripwire (string)
+		return original || player.getMainHandStack().isIn(ConventionalItemTags.SHEARS);
+	}
```

Comment:
Use a `@WrapOperation` instead of capturing locals, and especially instead of recomputing the receiver.